snapshot.go

```diff
@@ -66,6 +69,33 @@ const (
 	minReadVersion = snapshotVersion
 )
 
+type snapshotMetrics struct {
+	snapshotsTotal            *prometheus.CounterVec
+	snapshotFileSizeBytes     prometheus.Gauge
+	snapshotDurationHistogram prometheus.Histogram
+}
+
+func newSnapshotMetrics(reg prometheus.Registerer) *snapshotMetrics {
+	reg = prometheus.WrapRegistererWithPrefix("frostdb_snapshots_", reg)
+	return &snapshotMetrics{
+		snapshotsTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "total",
+			Help: "Total number of snapshots",
+		},
+			[]string{"success"},
+		),
+		snapshotFileSizeBytes: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "file_size_bytes",
+			Help: "Size of snapshots in bytes",
+		}),
+		snapshotDurationHistogram: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:    "duration_seconds",
+			Help:    "Duration of snapshots in seconds",
+			Buckets: prometheus.ExponentialBucketsRange(0.1, 120, 30),
```

Comment:
That's a lot of buckets for a metric that we have per database, we might want to adapt this after we get a feeling for the snapshot durations in prod.

The number of series we get out of this is `30 * number of databases * nodes in the system`, so even now this is already ~1000 series. It's negligible for now but something to keep in mind.